### DISCUSSION TITLE
Create a way to specify dependencies and packages on a project folder without Meson

### DISCUSSION THREAD

**Discussion by darltrash** [Prince781/vala-language-server#307] [Ideas]
Something like clangd's `compiler_flags.txt` which is literally just a list of compiler flags, it could even use the same code for the shebang parser.

The difference between this and the shebang is that this would apply to all the files within the project folder and subfolders, instead of per file.

**Comment by darltrash** [comment]
Apparently `compile_commands.json` exists, but isnt it meant to always be autogenerated?

**Reply by Prince781** [reply]
In theory you could hand-write your own to work with VLS. We have a backend just for `compiler_commands.json`. Often times this is what we fallback to when the Meson configuration fails, since if the user created a Meson build directory in the project it will contain a compile commands file.

**Reply by darltrash** [reply]
Yeah I was looking at the source, looks easy enough to do, what name should I give it though?
